pkg/timeutil/window_test.go

```diff
@@ -0,0 +1,66 @@
+package timeutil
+
+import (
+	"testing"
+	"time"
+)
+
+func TestWindowParse(t *testing.T) {
+	ptest := []struct {
+		start     string
+		end       string
+		shouldErr bool
+	}{
+		{"", "", true},
+		{"Sat 1:00", "2:00", true},
+		{"1:00:00", "2:00:00", true},
+		{"foo bar baz", "quux", true},
+		{"Sat 16:00", "60", false},
+		{"Saturday 16:00", "60", true},
+		{"Foo 16:00", "60", true},
+		{"Sat 4:00pm", "60", true},
+		{"foo:00pm", "60", true},
+		{"15:foo", "15", true},
+		{"Sat 16:00", "-1", true},
+	}
+
+	for i, tt := range ptest {
+		_, err := ParseWindow(tt.start, tt.end)
+		if (err != nil) != tt.shouldErr {
+			t.Errorf("test %d failed: %s", i, err)
+		}
+	}
+}
+
+func TestWindowNext(t *testing.T) {
+	ntest := []struct {
+		start  string
+		length string
+		time   time.Time
+		next   time.Duration
+	}{
+		{"00:00", "60", time.Date(2015, time.May, 15, 0, 0, 0, 0, time.Local), 0},
+		{"00:00", "60", time.Date(2015, time.May, 15, 1, 0, 0, 0, time.Local), 23 * time.Hour},
+		{"01:00", "60", time.Date(2015, time.May, 15, 0, 0, 0, 0, time.Local), 60 * time.Minute},
+		{"01:00", "60", time.Date(2015, time.May, 15, 23, 0, 0, 0, time.Local), 2 * time.Hour},
+
```

Comment:
Something like: `{"23:00", "120", time.Date(2015, time.May, 15, 0, 0, 0, 0, time.Local), 0},`. Where the reboot window itself wraps around midnight.